pkg/sessions/redis/redis_store.go

```diff
@@ -292,6 +250,67 @@ func (store *SessionStore) storeValue(ctx context.Context, value string, expirat
 	return ticket.encodeTicket(store.Cookie.Name), nil
 }
 
+// loadSessionFromTicket loads the session based on the ticket value
+func (store *SessionStore) loadSessionFromTicket(ctx context.Context, value string) (*sessions.SessionState, error) {
+	ticket, err := decodeTicket(store.Cookie.Name, value)
+	if err != nil {
+		return nil, err
+	}
+
+	resultBytes, err := store.Client.Get(ctx, ticket.asHandle(store.Cookie.Name))
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := encryption.NewGCMCipher(ticket.Secret)
+	if err != nil {
+		return nil, err
+	}
+
+	session, err := sessions.DecodeSessionState(resultBytes, c, false)
+	if err != nil {
+		// The GCM cipher will error due to a legacy JSON payload not passing
+		// the authentication check part of AES GCM encryption.
+		// In that case, we can attempt to fallback to try a legacy load
+		legacyCipher := encryption.NewBase64Cipher(store.CookieCipher)
+		return LegacyV5DecodeSession(resultBytes, ticket, legacyCipher)
+	}
+	return session, nil
+}
+
+// LegacyV5DecodeSession loads the session based on the ticket value
+// This fallback uses V5 style encryption of Base64 + AES CFB
+func LegacyV5DecodeSession(resultBytes []byte, ticket *TicketData, c encryption.Cipher) (*sessions.SessionState, error) {
```

Comment:
Does this method need to be public?

Reply:
Yeah - it actually isn't used internally in this package at all -- It is consumed by the cookie/redis session stores when they encounter a legacy  session.

Reply:
If it is used by both cookie and redis, should it be in a utils package or something rather than the cookie package importing the redis package?

Reply:
Whoops, I mixed this up with the sessions.LegacyV5DecodeSession -- this one can be private, good call!